web/src/api/hfiCalculatorAPI.ts

```diff
@@ -71,3 +77,43 @@ export async function getDailies(
     last_updated: DateTime.fromISO(daily.last_updated)
   }))
 }
+
+export async function getHFIResult(
+  request: HFIResultRequest
+): Promise<HFIResultResponse> {
+  const { data } = await axios.post<RawHFIResultResponse>(baseUrl, {
+    ...request,
+    selected_prep_date: request.selected_prep_date?.toISOString().split('T')[0] // Infinite sadness
+  })
+
+  data.planning_area_hfi_results.map(areaResult =>
+    areaResult.daily_results.map(dailyResult => dailyResult.dateISO)
+  )
+
+  const planningAreaResultsWithDates: PlanningAreaResult[] =
+    data.planning_area_hfi_results.map(areaResult => ({
+      ...areaResult,
+      daily_results: areaResult.daily_results.map(dr => ({
+        ...dr,
+        dailies: dr.dailies.map(validatedDaily => ({
+          ...validatedDaily,
+          daily: {
+            ...validatedDaily.daily,
+            date: DateTime.fromISO(
+              validatedDaily.daily.date.split('T')[0] + 'T' + '00:00-08:00' // Infinite sadness
```

Comment:
Same as above - re function and comment (except this is re-used, so function makes more sense). Also the 8 hour offset we should be using a constant.

Reply:
all that nested mapping is infinite sadness too :(